### ISSUE TITLE
Simply won't install

### ISSUE THREAD

**Issue by Stealthbird97** [castlamp/zenbership#95]
Hi,
I'm a little lost for ideas.
I simply can not get Zenbership to install.

I have been following the install guide http://documentation.zenbership.com/Basics/Installation-and-Setup
I access the setup page, chmod the folders so that everything is green.
GD shows as red (however I do have that installed so why it is red I don't know).

I enter the MySQL DB info and test, and I get "success".
Edit the other information and finally click "Process Setup", and it will just hang for ages and ages.
If I click Process Setup a second time, it just shows a load of php. [process.php.txt](https://github.com/castlamp/zenbership/files/1077514/process.php.txt)
I've check my access and error logs which flag up nothing.
I've tried with fastcgi, php-fpm for 5.6.30 and 7.0.

Something I have noticed is that if I do not test the MySQL info before I click Process Setup, I just get a load of php.
[process.php.txt](https://github.com/castlamp/zenbership/files/1077514/process.php.txt)

[error.log.txt](https://github.com/castlamp/zenbership/files/1077519/error.log.txt)
Given it isn't setup and so the config.php file does not exist, I would expect this error but it doesn't explain why everything else does not work...

Can anyone offer suggestions?

System:
Ubuntu 16.04
Apache2, php7, php5.6.30
ISPConfig

http://members.radio-noise.net/setup/

**Comment by prodigy151** [comment]
I had the same issue. GD should be green. Make sure you have installed it. I noticed that the site does not work on php 7. I had to install php 5.6 and enable it. Here is what I did:

--Install php 5.6
sudo add-apt-repository ppa:ondrej/php
sudo apt-get update
sudo apt-get upgrade
sudo apt-get install php5.6 

--Install all libraries and extensions including GD
sudo apt-get install php5.6-gd php5.6-mysql php5.6-dom php5.6-cli php5.6-json php5.6-common php5.6-mbstring php5.6-opcache php5.6-readline php5.6-curl php5.6-mcrypt

--Make sure your mysql database is set to traditional mode. This has to be done before you start the installation process. You need to update the MySQL Config File: /etc/mysql/my.cnf file and add the following:
[mysqld]
sql_mode = "IGNORE_SPACE,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"

--Enable php 5.6 and make sure it is active
--From php7.0 to php5.6:
sudo a2dismod php7.0 ; sudo a2enmod php5.6 ; sudo service apache2 restart


here is a link if you want to reinstall php 5.6 on your system.

Hope this helps.

**Comment by jlrcontegix** [comment]
I'm not sure if it's just an issue at install/setup time, but I've been using PHP 7.0/7.1 with several Zenbership releases without issue.

**Comment by prodigy151** [comment]
The install did not work for me on php 7. For some reason, it did really well on php 5.6.

**Comment by Stealthbird97** [comment]
@prodigy151, Unfortunately, I have already done everything you have suggested.
I already have php5.6 installed and GD is enabled...
http://members.radio-noise.net/setup/phpinfo.php

**Comment by Stealthbird97** [comment]
To add, I have also attempted using a different host, exactly the same error..
http://member.solentoverlord.org.uk/setup/phpinfo.php

**Comment by Stealthbird97** [comment]
@jbelelieu @castlamp Can you shine any light here? As far as I can see, I have done everything stated in the setup guide, and my servers meet the requirements. Why does it not work?

**Comment by Stealthbird97** [comment]
Well, this is very disappointing...
This software was exactly what I needed but it doesn't work and no solution in site.

**Comment by dsnyder0li** [comment]
@Stealthbird97 - have you had any luck getting your install to work? I was able to automate the installation using Saltstack. I can now go from a base Ubuntu 16.04 LTS image either in VirtualBox or Linode to a working Zenbership installation in about six minutes.

I found that I had to give the webserver user (www-data in my case) write access to the following subdirs after cloning the Zenbership repo (note that backups and backups/db need to be created):

`  - admin/sd-system
  - admin/sd-system/attachments
  - admin/sd-system/exports
  - admin/sd-system/backups
  - admin/sd-system/backups/db
  - custom/sessions
  - custom/qrcodes
  - custom/uploads`

After doing that, I used a script like this to configure Zenbership:

`#!/usr/bin/env python
import requests

setup_data = {
  "mysql[host]": "localhost",
  "mysql[db]": "zenbership",
  "mysql[user]": "zendbuser",
  "mysql[pass]": "seCret_passw0rd",
  "company_name": "My Company",
  "company_url": "https://example.com",
  "company_email": "sales@example.com",
  "company_logo": "https://example.com/images/logo.png",
  "site_name": "Awesome Members",
  "hour_change": "-8",
  "admin[first_name]": "Mr",
  "admin[last_name]": "Administrator",
  "admin[email]": "support@example.com",
  "admin[user]": "admin",
  "admin[pass]": "mY_sUpser_Pass",
  "enroll_newsletter": "1",
  "enroll_stats": "1",
}
response = requests.post("https://example.com/setup/process.php", data = setup_data, verify = "/etc/ssl/certs/cert.pem")
response.raise_for_status()
print response.status_code`

Remove the setup subir, fix permissions on the admin/cp-cron files, and add the cron jobs and you should be all set assuming that the rest of your environment is good.

Oh, before doing any of this, I made sure that I had the right O/S packages installed. Here's a list (not all are required for Zenbership):

`      - mysql-common
      - mysql-server
      - php-mysql
      - python-mysqldb
      - nginx
      - php-fpm
      - php-mcrypt
      - php-curl
      - python-certbot-nginx
      - sudo
      - ntpdate
      - ntp
      - git
      - openssl
      - vile
      - nano
      - screen
      - tmux
      - jq
      - apg
      - unzip
      - zip
      - mailutils
      - postfix`

I hope this helps.

**Comment by rmccaskill** [comment]
REALLY wish there was some more direction / specifics around the install. I'm already on 3 different VMs with multiple versions of OSs and VMs installed...  I almost had v115 going with Ubuntu 16, and php5.6 but had some issues with forms not being editable and cron tasks not working. Trying again from scratch...

Actually, I'm wondering if someone would make a Vagrant image or something that 'just works'.

**Comment by dsnyder0li** [comment]
No promises, but if I have time to make my Salt states "beautiful", I may put them into a public repo. Between now and then, let me know if you have any specific questions.

**Comment by vsilent** [comment]
I started https://github.com/trydirect/zenbership  , I get it working with php 7.3.  The only thing remains is NGINX configuration
